packages/@aws-cdk/aws-glue-alpha/lib/storage-parameter.ts

```diff
@@ -0,0 +1,323 @@
+   */
+  TRUNCATE = 'TRUNCATE',
+}
+
+/**
+ * The action to assign to `COLUMN_COUNT_MISMATCH_HANDLING`.
+ */
+export enum SurplusCharHandlingAction {
+  /**
+   * Replaces data that exceeds the column width with null.
+   */
+  SET_TO_NULL = 'SET_TO_NULL',
+
+  /**
+   * Doesn't perform surplus character handling.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * Cancels queries that return data exceeding the column width.
+   */
+  FAIL = 'FAIL',
+
+  /**
+   * Replaces each value in the row with null.
+   */
+  DROP_ROW = 'DROP_ROW',
+
+  /**
+   * Removes the characters that exceed the maximum number of characters defined for the column.
+   */
+  TRUNCATE = 'TRUNCATE',
+}
+
+/**
+ * The action to assign to `COLUMN_COUNT_MISMATCH_HANDLING`.
+ */
+export enum ColumnCountMismatchHandlingAction {
+  /**
+   * Column count mismatch handling is turned off.
+   */
+  DISABLED = 'DISABLED',
+
+  /**
+   * Fail the query if the column count mismatch is detected.
+   */
+  FAIL = 'FAIL',
+
+  /**
+   * Fill missing values with NULL and ignore the additional values in each row.
+   */
+  SET_TO_NULL = 'SET_TO_NULL',
+
+  /**
+   * Drop all rows that contain column count mismatch error from the scan.
+   */
+  DROP_ROW = 'DROP_ROW',
+}
+
+export class StorageParameterValue {
+  /**
+   * Assigns a value of `CompressionType`
+   */
+  public static compressionType(value: CompressionType): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * Assigns a value of `InvalidCharHandlingAction`
+   */
+  public static invalidCharHandling(value: InvalidCharHandlingAction): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * Assigns a value of `NumericOverflowHandlingAction`
+   */
+  public static numericOverflowHandling(value: NumericOverflowHandlingAction): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * Assigns a value of `SurplusBytesHandlingAction`
+   */
+  public static surplusBytesHandling(value: SurplusBytesHandlingAction): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * Assigns a value of `SurplusCharHandlingAction`
+   */
+  public static surplusCharHandling(value: SurplusCharHandlingAction): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * Assigns a value of `ColumnCountMismatchHandlingAction`
+   */
+  public static columnCountMismatchHandling(value: ColumnCountMismatchHandlingAction): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  /**
+   * A custom storage parameter value that can be used to assign to any storage parameter.
+   * @param value - The value of the storage parameter.
+   */
+  public static custom(value: string): StorageParameterValue {
+    return new StorageParameterValue(value);
+  }
+
+  protected constructor(public readonly value: string) {}
+}
+
+export class StorageParameters {
```

Comment:
Do you have a documentation reference for all of these?

Specifically I'm curious if they can always be applied or are only valid in certain circumstances.

Reply:
Hi, I have not got documentation reference for these specific values, other than comments within the code. Should I include these within the README?
Also, they are only applicable for the specified keys.

Reply:
Gotcha. I'm mostly trying to get an understanding of the problem space right now.
E.g. how does a user know they can set the compression parameter?

Reply:
For questions referring to the type of "How does the user know x,y,z?", it'd mostly be through the README. I think the best course of action would be to set out a table like structure within the README, to help navigate the different variables, just like the example given for [primitives](https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/aws-glue-alpha#primitives)?

Reply:
That would be a good idea (once the rest of the PR is okay).  But what is the _source_ of the table and available parameters? We should have `@see` referenced to AWS docs, and not rely on contributor's knowledge of Glue.

Reply:
The location of this documentation has been added. There is no perma-link to the sub heading "TABLE PROPERTIES", so I just added this after the link; to let users know.